src/model/NPMResolver.ts

```diff
@@ -0,0 +1,194 @@
+/*
+ * Copyright 2018, collaboration Factory AG. All rights reserved.
+ */
+
+import * as path from "path";
+import * as fs from "fs";
+import * as crypto from "crypto";
+import * as spawn from 'cross-spawn';
+import * as chokidar from "chokidar";
+import {Scheduler} from "../executor";
+import {cerr, cgreen, cred} from "../utils";
+import Timeout = NodeJS.Timeout;
+import * as rimraf from "rimraf";
+
+export class NPMResolver {
+    private static readonly PACKAGE_LOCK_HASH = 'package-lock.hash';
+    private static readonly PACKAGE_LOCK_JSON = 'package-lock.json';
+    private static readonly PACKAGE_JSON = 'package.json';
+    private static readonly NODE_MODULES = 'node_modules';
+    private mainRepo: string;
+    private readonly hashFilePath: string;
+
+    constructor(mainRepo: string, watch: boolean, clean: boolean) {
+        this.mainRepo = mainRepo;
+        if (clean) {
+            this.clean();
+        }
+        this.hashFilePath = this.getHashFilePath();
+        if (!this.isNPMInstallVersion()) {
+            console.log(cgreen`⇢` + ` [NPM] package.json version is 1.0.0 -> no npm install needed / your node modules are checked in`);
+        } else {
+            console.log(cgreen`⇢` + ` [NPM] package.json version is 2.0.0 or higher -> checking if npm install is needed`);
+            this.checkAndInstall();
+            if (watch) {
+                this.registerWatchers();
+            }
+        }
+    }
+
+    private isNPMInstallVersion(): boolean {
+        return !(this.getPackageVersion() === '1.0.0');
+    }
+
+    private clean() {
```

Comment:
Since it's not possible to install the node_modules without internet connection we should not remove them when a user does a clean - which happens very often. Otherwise they suddenly would remove their own base for working...